test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+                    // these points at +180 and the polygon part in the west hemisphere should have these points at -180. so we will determine
+                    // how many of the polygon part vertices are in the positive hemisphere and how many are in the negative hemisphere and also
+                    // exclude the dateline points. if most are negative then make the dateline coordinate -180 else make it +180.
+                    var coordsInPos = geomCoords.FindAll(c => c.X >= 0 & Math.Abs(c.X) != 180);
+                    var coordsInNeg = geomCoords.FindAll(c => c.X < 0 & Math.Abs(c.X) != 180);
+                    var coordsAtDtl = geomCoords.FindAll(c => Math.Abs(c.X) == 180);
+                    if (coordsInPos.Count >= coordsInNeg.Count)
+                    {
+                        coordsAtDtl.ForEach(c => c.X = 180);
+                    }
+                    else
+                    {
+                        coordsAtDtl.ForEach((c) => c.X = -180);
+                    }
+
+
+                    // print to debug which hemisphere this geometry is in since it doesnt cross the dateline
+                    string pfx = string.Format("unshifted geometry part {0} is in ", w);
+                    DebugPrintHemi(geomCoords, pfx);
+                }
+            }
+
+            // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+            // here we will densify the polygon if the resolution value is passed in.
+            // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+            if (!double.IsNaN(outDensifyResolution) && !double.IsInfinity(outDensifyResolution) && outDensifyResolution > 0d)
+            {
+                resGeoms = Densifier.Densify(resGeoms, outDensifyResolution);
+            }
+
+            // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+            // finally if the option is passed in to attempt to fix invalid geometry then we will do that here
+            // https://github.com/NetTopologySuite/NetTopologySuite/issues/124
+            // Attempting to fix an invalid geometry by calling geometry.Buffer(0) is a common approach, and
+            // suggested in multiple places. However it does not work in all cases.
+            // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+            if (isAttFixOutInvPolygons)
+            {
+                var valOp = new IsValidOp(resGeoms);
+                Debug.Print("resGeoms isValid = {0}", valOp.IsValid);
+                if (!valOp.IsValid)
+                {
+                    // method 1
+                    resGeoms = resGeoms.Buffer(0);
+                    Debug.Print("resGeoms were fixed");
+
+                    // method 2
+                    //var pm = new PrecisionModel(10000000000.0);
```

Comment:
Old code that can be removed? or it can be handled using a specific methofd parameter?

Reply:
well no. the routine to fix invalid geometries resulted from some research and eventually this article: https://github.com/NetTopologySuite/NetTopologySuite/issues/124
From that discussion it was noted that Buffer does not always work. So I added the second method in case I wanted to  add as an option later. I could probably make the input an Enum rather than a Boolean. So I would have 

```
    public enum attFixOutInvPolygons: int
    {
        FixNone= 0,
        FixViaBuffer=1,
        FixViaPrecision=2,
    }
```

Reply:
I like the enum idea

Reply:
Im adding it as my next commit.